web/ui/react-app/src/pages/Flags.tsx

```diff
@@ -1,51 +1,41 @@
-import React, { FC } from 'react';
+import React from 'react';
 import { RouteComponentProps } from '@reach/router';
 import { Alert, Table } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faSpinner } from '@fortawesome/free-solid-svg-icons';
-import { useFetch } from '../utils/useFetch';
+import { Fetch, FetchState } from '../api/Fetch';
 import PathPrefixProps from '../PathPrefixProps';
 
 export interface FlagMap {
   [key: string]: string;
 }
 
-const Flags: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
-  const { response, error } = useFetch(`${pathPrefix}/api/v1/status/flags`);
-
-  const body = () => {
-    const flags: FlagMap = response && response.data;
-    if (error) {
-      return (
-        <Alert color="danger">
-          <strong>Error:</strong> Error fetching flags: {error.message}
-        </Alert>
-      );
-    } else if (flags) {
-      return (
-        <Table bordered={true} size="sm" striped={true}>
-          <tbody>
-            {Object.keys(flags).map(key => {
-              return (
-                <tr key={key}>
-                  <th>{key}</th>
-                  <td>{flags[key]}</td>
-                </tr>
-              );
-            })}
-          </tbody>
-        </Table>
-      );
-    }
-    return <FontAwesomeIcon icon={faSpinner} spin />;
-  };
-
-  return (
-    <>
-      <h2>Command-Line Flags</h2>
-      {body()}
-    </>
-  );
-};
+export const Flags = ({ error, data: flags }: FetchState<FlagMap>) => {
```

Comment:
Thinking more, I see value in being as explicit as possible and just making it a full functional component, like:

```suggestion
export const Flags: FC<FetchState<FlagMap>> = ({ error, data: flags }) => {
```

Fewer linter warnings about undefined return values, and clearer to the reader that it is (or can be used) as a component.

I'd do the same typing below wrapped component, though I don't know how you can do the `FC<...>` typing inline after `export default`. But assigning it to another name first is cool too.

For naming, maybe `RenderFlags` and for the exported one, just `Flags`?